docs/interactions/Application_Commands.md

```diff
@@ -424,183 +638,317 @@ And, done! The JSON looks a bit complicated, but what we've ended up with is a s
 
 ![A command with grouped subcommands and parameters. It says /permissions user get with arguments for a user and a channel.](command-with-groups-subcommands-parameters.png)
 
-## Permissions
-
-Need to keep some of your Slash Commands safe from prying eyes, or only available to the right people? Slash Commands support permission overwrites for all your commands. For both guild _and_ global commands, you can enable or disable a specific user or role in a guild from using a command.
+## User Commands
 
-> info
-> For now, if you don't have permission to use a command, they'll show up in the command picker as disabled and unusable. They will not be hidden.
+User commands are application commands that appear on the context menu (right click or tap) of users. They're a great way to surface quick actions for your app that target users. They don't take any arguments, and will return the user on whom you clicked or tapped in the interaction response.
 
-You can also set a `default_permission` on your commands if you want them to be disabled by default when your app is added to a new guild. Setting `default_permission` to `false` will disallow _anyone_ in a guild from using the command--even Administrators and guild owners--unless a specific overwrite is configured. It will also disable the command from being usable in DMs.
+> danger
+> The `description` field is not allowed when creating user commands. However, to avoid breaking changes to data models, `description` will be an **empty string** (instead of `null`) when fetching commands.
 
-For example, this command will not be usable by anyone in any guilds by default:
+###### Example User Command
 
 ```json
 {
-    "name": "permissions_test",
-    "description": "A test of default permissions",
-    "default_permission": false
+    "name": "High Five",
+    "type": 2
 }
 ```
 
-To enable it just for a moderator role:
+![An example user command. The context menu has an Apps section open to a High Five command](message-command.png)
 
-```py
-MODERATOR_ROLE_ID = "<moderator_role_id>"
-url = "https://discord.com/api/v8/applications/<my_application_id>/guilds/<my_guild_id>/commands/<my_command_id>/permissions"
+When someone uses a user command, your application will receive an interaction:
 
-json = {
-    "permissions": [
-        {
-            "id": MODERATOR_ROLE_ID,
-            "type": 1,
-            "permission": True
+###### Example Interaction
```

Comment:
Way to differentiate between slash cmds and context menu cmds is by keeping track of the command id?

Reply:
Or their type.

`1`: Slash command
`2`: User command
`3`: Message command